linkerd/app/gateway/src/lib.rs

```diff
@@ -23,30 +22,37 @@ use linkerd_app_inbound::{
 use linkerd_app_outbound::{self as outbound, Outbound};
 use std::fmt;
 use thiserror::Error;
-use tracing::debug_span;
 
 #[derive(Clone, Debug, Default)]
 pub struct Config {
     pub allow_discovery: NameMatch,
 }
 
 #[derive(Clone, Debug)]
-struct HttpTransportHeader {
-    target: NameAddr,
+struct HttpWithClient {
+    http: Http,
     client: ClientInfo,
-    version: http::Version,
-    policy: policy::AllowPolicy,
+    inbound_policy: policy::AllowPolicy,
 }
 
 #[derive(Clone, Debug)]
-struct RouteHttp<T>(T);
-
-#[derive(Clone, Debug, PartialEq, Eq, Hash)]
-struct HttpTarget {
+struct Http {
```

Comment:
it could _maybe_ be worth adding comments to these structs explaining their relationship, for future readers? Something like "`HttpWithClient` is the target for the HTTP server stack, and then the client info is removed from the target and added to a request extension, and services are cached based on the `Http` target"?